src/test/java/com/commercetools/sync/taxcategories/TaxCategorySyncOptionsTest.java

```diff
@@ -0,0 +1,158 @@
+package com.commercetools.sync.taxcategories;
+
+import com.commercetools.sync.commons.utils.TriFunction;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.commands.UpdateAction;
+import io.sphere.sdk.taxcategories.TaxCategory;
+import io.sphere.sdk.taxcategories.TaxCategoryDraft;
+import io.sphere.sdk.taxcategories.TaxCategoryDraftBuilder;
+import io.sphere.sdk.taxcategories.commands.updateactions.SetKey;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class TaxCategorySyncOptionsTest {
+
+    private static SphereClient CTP_CLIENT = mock(SphereClient.class);
+
+    @Test
+    void applyBeforeUpdateCallback_WithNullCallback_ShouldReturnIdenticalList() {
+        final TaxCategorySyncOptions taxCategorySyncOptions = TaxCategorySyncOptionsBuilder.of(CTP_CLIENT)
+            .build();
+        assertThat(taxCategorySyncOptions.getBeforeUpdateCallback()).isNull();
+
+        List<UpdateAction<TaxCategory>> updateActions = singletonList(SetKey.of("key"));
+        List<UpdateAction<TaxCategory>> filteredList = taxCategorySyncOptions
+            .applyBeforeUpdateCallBack(updateActions, mock(TaxCategoryDraft.class), mock(TaxCategory.class));
+
+        assertThat(filteredList).as(" returned 'updateActions' should not be changed")
```

Comment:
This `as` description is supposed to be a description for the assertion not the actual expectation, because if it fails, it usually prints the expectation and actual values. For example: 
`"[check Frodo's age] expected:<[33]> but was:<[50]>"`




```suggestion
        assertThat(filteredList).as("check list of update actions after applying beforeUpdateCallback")
```
-> example of failing test:
````java
java.lang.AssertionError: [check list of update actions after applying beforeUpdateCallback] 
Expecting:
 <null>
and actual:
 <[SetKey[key=key,action=setKey]]>
to refer to the same object
````